### ISSUE TITLE
brew search suggests installing a cask

### ISSUE THREAD

**Issue by bukzor** [Homebrew/linuxbrew#284]
I was just trying to see if linuxbrew would help install virtualbox. I followed the on-screen suggestions, but they lead in a circle.

```
$ brew search virtualbox
Caskroom/cask/virtualbox
$ brew info Caskroom/cask/virtualbox
Error: No available formula for virtualbox 
Please tap it and then try again: brew tap Caskroom/cask
$ brew tap Caskroom/cask
Warning: Already tapped!
$ brew info Caskroom/cask/virtualbox
Error: No available formula for virtualbox 
Please tap it and then try again: brew tap Caskroom/cask
$ brew info cask/virtualbox
Error: No available formula for cask/virtualbox 
$ brew info virtualbox
Error: No available formula for virtualbox 
```

**Comment by Pfeil** [comment]
Because brew on mac only provides terminal applications, there is an extension called cask to install sandboxed mac applications. I'm not sure whether cask wants to offer binary applications for other systems. Anyway, on my Mac I can view the formula, and it only provides a .dmg for OSX:

    Pfeil ~> brew cask cat virtualbox
    cask :v1 => 'virtualbox' do
      version '4.3.22-98236'
      sha256 '886d0e6d17ee09cc1f48d7319257d597ed35428237a4d0d2305a7e03347b717d'

      url "http://download.virtualbox.org/virtualbox/#{version.sub(/-.*$/, '')}/VirtualBox-#{version}-OSX.dmg"
      name 'VirtualBox'
      homepage 'http://www.virtualbox.org'
      license :gpl
      tags :vendor => 'Oracle'
    
      pkg 'VirtualBox.pkg'
      binary '/Applications/VirtualBox.app/Contents/MacOS/VBoxManage'
      binary '/Applications/VirtualBox.app/Contents/MacOS/VBoxHeadless'
    
      uninstall :script => { :executable => 'VirtualBox_Uninstall.tool', :args => %w[--unattended] },
                :pkgutil => 'org.virtualbox.pkg.*'
    end
    Pfeil ~>

See the url string. Casks exist because of the way graphical applications are packaged on OSX (one container (name.app) with the binaries etc inside, doubleclick on container starts application). On Linux they do not make too much sense. You would need to create a formula in the linuxbrew repository to install virtuallbox using brew, I guess.

I think it would make sense to remove the cask repository for linuxbrew users if possible. The behavior of linuxbrew, though is totally normal. It's the same that homebrew for mac shows. Even on OSX you need to install brew-cask to use the `brew cask` command.

**Comment by bukzor** [comment]
I was using 'brew info'.

**Comment by Pfeil** [comment]
Yes. I have no idea why homebrew and linuxbrew show casks without brew-cask being installed (but I can confirm it), but they are OSX-specific. More information to cask: https://github.com/caskroom/homebrew-cask

> Homebrew Cask extends Homebrew and brings its elegance, simplicity, and speed to OS X applications and large binaries alike. (http://caskroom.io/)

I think the only issue here is that brew shows casks by default. Especially on linux.

**Comment by sjackman** [comment]
Yes, I agree that `brew search` should not show casks on Linux.

**Comment by bukzor** [comment]
It's brew info that's behaving strangely, and I imagine it's the same for
all taps. I don't think the issue is about casks per se.

--phone is hard.

**Comment by Pfeil** [comment]
Do you have problems with non-cask repositories? If not, I think this is normal. There is a technical difference between formula and casks. Formulas derive from "Formula":

    $ brew cat fish
    class Fish < Formula
    ...
    end

Casks do not even seem to be a class (I don't know this construct, I'm no ruby programmer):

    $ brew cask cat virtualbox
    cask :v1 => 'virtualbox' do
    ...
    end

And since virtualbox is (in this case) no formula, but a cask, `brew info` does not find any (I guess).

    $ brew info Caskroom/cask/virtualbox
    Error: No available formula for virtualbox

**Comment by bukzor** [comment]
Ah, you're right! This works exactly the same in osx.

My mistake.

On Wed, Feb 25, 2015 at 10:29 AM, Andreas Pfeil <notifications@github.com>
wrote:

> Do you have problems with non-cask repositories? If not, I think this is
> normal. There is a technical difference between formula and casks. Formulas
> derive from "Formula":
>
> $ brew cat fish
> class Fish < Formula
> ...
> end
>
> Casks do not even seem to be a class (I don't know this construct, I'm no
> ruby programmer):
>
> $ brew cask cat virtualbox
> cask :v1 => 'virtualbox' do
> ...
> end
>
> And since virtualbox is (in this case) no formula, but a cask, brew info
> does not find any (I guess).
>
> $ brew info Caskroom/cask/virtualbox
> Error: No available formula for virtualbox
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Homebrew/linuxbrew/issues/284#issuecomment-76024633>.
>



-- 
Buck Golemon
Cell: (408) 202-3901

**Comment by bukzor** [comment]
That's really what I was complaining about in the OP.
You can close this as NOTABUG if you like.

On Wed, Feb 25, 2015 at 10:52 AM, Buck Evan <workitharder@gmail.com> wrote:

> Ah, you're right! This works exactly the same in osx.
>
> My mistake.
>
> On Wed, Feb 25, 2015 at 10:29 AM, Andreas Pfeil <notifications@github.com>
> wrote:
>
>> Do you have problems with non-cask repositories? If not, I think this is
>> normal. There is a technical difference between formula and casks. Formulas
>> derive from "Formula":
>>
>> $ brew cat fish
>> class Fish < Formula
>> ...
>> end
>>
>> Casks do not even seem to be a class (I don't know this construct, I'm no
>> ruby programmer):
>>
>> $ brew cask cat virtualbox
>> cask :v1 => 'virtualbox' do
>> ...
>> end
>>
>> And since virtualbox is (in this case) no formula, but a cask, brew info
>> does not find any (I guess).
>>
>> $ brew info Caskroom/cask/virtualbox
>> Error: No available formula for virtualbox
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/Homebrew/linuxbrew/issues/284#issuecomment-76024633>.
>>
>
>
>
> --
> Buck Golemon
> Cell: (408) 202-3901
>
>


-- 
Buck Golemon
Cell: (408) 202-3901

**Comment by sjackman** [comment]
It's still a bug, just not a high priority one. Linuxbrew should not suggest installing a cask, which is not available for Linux.

**Comment by iggyvolz** [comment]
This should have been closed by #299.

**Comment by sjackman** [comment]
Fixed by #299 and https://github.com/Homebrew/linuxbrew/commit/8fe17ee3ae3cd158804193b9e28e42ce29e19b6c
